src/datasources/entity-ds-react/EntityDataSource.ts

```diff
@@ -44,10 +45,27 @@ export class EntityDataSource extends DataSourceApi<EntityQuery> {
     async query(request: EntityQueryRequest<EntityQuery>): Promise<DataQueryResponse> {
         const fullData: OnmsTableData[] = []
 
+        // TODO:
```

Comment:
Yeah this solution is definitely not ideal, the use of localStorage, but this whole filter panel is a hack to begin with, so having to use further hacks to make it work only makes sense.

Another option I thought about a week or so ago for this problem, would be to store the relevant JSON data as a JSON.stringify() string within the DOM. Maybe with an ID so it's easy to retrieve. Then we can just parse that JSON string to get the data, and it would ensure that the Filter panel data is always tied to the right dashboard, and that the data goes away when deleting the panel, AND it means that a user can wipe their localstorage and still get their filter panel back.

Just a thought. :)